usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,184 @@
+    },
+    "subaward": {
+        "minimum_db_fields": {"subaward_number", "piid", "fain", "award_type"},
+        "api_to_db_mapping_list": [contract_subaward_mapping, grant_subaward_mapping],
+        "award_semaphore": "award_type",
+        "award_id_fields": ["award__piid", "award__fain"],
+        "internal_id_field": "subaward_number",
+        "type_code_to_field_map": {"procurement": contract_subaward_mapping, "grant": grant_subaward_mapping},
+        "filter_queryset_func": subaward_filter,
+    },
+}
+
+
+@api_transformations(api_version=settings.API_VERSION, function_list=API_TRANSFORM_FUNCTIONS)
+class SpendingByAwardVisualizationViewSet(APIView):
+    """
+    This route takes award filters and fields, and returns the fields of the filtered awards.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/search/spending_by_award.md"
+
+    @cache_response()
+    def post(self, request):
+        """Return all awards matching the provided filters and limits"""
+        json_request = self.validate_request_data(request.data)
+        self.is_subaward = json_request["subawards"]
+        self.constants = GLOBAL_MAP["subaward"] if self.is_subaward else GLOBAL_MAP["award"]
+        self.filters = json_request["filters"]
+        self.fields = json_request["fields"]
+        self.pagination = {
+            "limit": json_request["limit"],
+            "lower_bound": (json_request["page"] - 1) * json_request["limit"],
+            "page": json_request["page"],
+            "sort_key": json_request.get("sort") or self.fields[0],
+            "sort_order": json_request["order"],
+            "upper_bound": json_request["page"] * json_request["limit"] + 1,
+        }
+
+        if self.if_no_intersection():  # Like an exception, but API response is a HTTP 200 with a JSON payload
+            return Response(self.populate_response(results=[], has_next=False))
+        raise_if_award_types_not_valid_subset(self.filters["award_type_codes"], self.is_subaward)
+        raise_if_sort_key_not_valid(self.pagination["sort_key"], self.fields, self.is_subaward)
+
+        return Response(self.create_response(self.construct_queryset()))
+
+    @staticmethod
+    def validate_request_data(request_data):
+        models = [
+            {"name": "fields", "key": "fields", "type": "array", "array_type": "text", "text_type": "search", "min": 1},
+            {"name": "subawards", "key": "subawards", "type": "boolean", "default": False},
+        ]
+        models.extend(copy.deepcopy(AWARD_FILTER))
+        models.extend(copy.deepcopy(PAGINATION))
+        for m in models:
+            if m["name"] in ("award_type_codes", "fields"):
+                m["optional"] = False
+
+        return TinyShield(models).block(request_data)
+
+    def if_no_intersection(self):
+        # "Special case" behavior: there will never be results when the website provides this value
+        return "no intersection" in self.filters["award_type_codes"]
```

Comment:
At one line, this was probably too small to be a function IMO

Reply:
Improves readability above. Devs can read the code and know exactly what the check is doing without caring about the details. The details are encapsulated in this function.

IMHO classes can be too small, but functions done properly should be small. Some well-respected developers recommend 2-4 lines. Including the comment (Because comments are lines) this is a 2-line function

Reply:
Converted, this would equate to `if "no intersection" in self.filters["award_type_codes"]:`. If that's too complicated to not be a function, why isn't every if statement?

I don't wish to disrespect some developers, but I probably made a mistake bringing line counts into this at all.

Reply:
At one point during this branch development, there were more lines in the function.

Because of repeated issues handling this special case, I would prefer to leave it as-is. I do realize the function is small and the logic is straight-forward. But the **significance** of the logic is not obvious. There have been numerous issues in the past with this value not being checked early enough in the view method and instead of a HTTP 200 response with the proper schema, a HTTP 500 is returned.

Reply:
I absolutely agree that something this silly should come first, and tbh I would actually rather let the function thing go and save my energy for the fight over the blank line. But I would much rather get through you reviewing MY PR before code freeze...